File: backend/src/main/java/com/java/NBE4_5_1_7/domain/community/post/dto/PostResponseDto.java
package com.java.NBE4_5_1_7.domain.community.post.dto;

import com.java.NBE4_5_1_7.domain.community.comment.dto.CommentResponseDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PostResponseDto {
    private Long id;
    private String authorName;
    private LocalDateTime postTime;
    private String title;
    private String content;
    private Integer like;
    private List<CommentResponseDto> comments;
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/community/post/dto/PostResponseDto.kt
package com.java.NBE4_5_3_7.domain.community.post.dto

import com.java.NBE4_5_3_7.domain.community.comment.dto.CommentResponseDto
import java.time.LocalDateTime

class PostResponseDto {
    var id: Long? = null
    var authorName: String? = null
    var postTime: LocalDateTime? = null
    var title: String? = null
    var content: String? = null
    var like: Int? = null
    var likedByCurrentUser: Boolean? = null
    var comments: List<CommentResponseDto>? = null


    constructor()

    constructor(
        id: Long?, authorName: String?, postTime: LocalDateTime?,
        title: String?, content: String?, like: Int?,
        likedByCurrentUser: Boolean?,
        comments: List<CommentResponseDto>?
    ) {
        this.id = id
        this.authorName = authorName
        this.postTime = postTime
        this.title = title
        this.content = content
        this.like = like
        this.likedByCurrentUser = likedByCurrentUser
        this.comments = comments
    }
}

